gr-vocoder/lib/freedv_tx_ss_impl.cc

```diff
@@ -91,17 +92,23 @@ void freedv_tx_ss_impl::set_tx_bpf(bool val)
         freedv_tx_ss_impl::set_tx_bpf(0);
 }
 
-int freedv_tx_ss_impl::work(int noutput_items,
-                            gr_vector_const_void_star& input_items,
-                            gr_vector_void_star& output_items)
+void freedv_tx_ss_impl::forecast(int noutput_items, gr_vector_int& ninput_items_required)
+{
+    ninput_items_required[0] = (noutput_items / d_nom_modem_samples) * d_speech_samples;
+}
+
+int freedv_tx_ss_impl::general_work(int noutput_items,
+                                    gr_vector_int& ninput_items,
+                                    gr_vector_const_void_star& input_items,
+                                    gr_vector_void_star& output_items)
 {
     short* in = (short*)input_items[0];
     short* out = (short*)output_items[0];
     int i;
 
     for (i = 0; i < (noutput_items / d_nom_modem_samples); i++)
-        freedv_tx(
-            d_freedv, &(out[i * d_nom_modem_samples]), &(in[i * d_nom_modem_samples]));
+        freedv_tx(d_freedv, &(out[i * d_nom_modem_samples]), &(in[i * d_speech_samples]));
+    consume(0, i * d_speech_samples);
```

Comment:
It looks like in other general blocks in the codebase with only 1 input, consume_each is always called

Reply:
Fixed.